DocNet.Console/Program.cs

```diff
@@ -1,15 +1,259 @@
 ﻿using log4net;
+using CommandLine;
 
 namespace DocNet.Console
 {
-    public class Program
+    using System;
+    using System.IO;
+
+    public static class Program
     {
+
         private static readonly ILog Log = LogManager.GetLogger(typeof(Program));
 
+        //Command Line Options
+        class Options
+        {
+            [Option('i', "input", Required = true, HelpText = "File/Folder to convert to documentation.")]
+            public string InputFile { get; set; }
+            [Option('o', "output", Required = true, HelpText = "Location to output documentation.")]
+            public string OutputFile { get; set; }
+            [Option('r', "recurse", HelpText = "Recursively search through input folder", DefaultValue = false)]
+            public bool RecurseOption { get; set; }
+            [Option('h', "help", HelpText = "Display command line help information", Required = false, DefaultValue = false)]
+            public bool Help { get; set; }
+        }
+
+        //Checks that command line input is correct and outputs a custom message depending on whats missing
+        static private bool CmdCheck(string[] args)
+        {
+            //Determine which required commands are missing
+            if (args.Length == 0)
+            {
+                Console.WriteLine("No commands found. Use the following commands:");
+                return false;
+            }
+            //Check if args contains --help
+            if ((Array.IndexOf(args, "-h") >= 0) || (Array.IndexOf(args, "--help") >= 0))
+            {
+                return false;
+            }
+            //Check if other commands exist
+            if (!(((Array.IndexOf(args, "-o") >= 0) || (Array.IndexOf(args, "--output") >= 0)) && ((Array.IndexOf(args, "-i") >= 0) || (Array.IndexOf(args, "--input") >= 0))))
+            {
+                Console.WriteLine("Required commands not found. Use the following commands:");
+                return false;
+            }
+            else
+            {
+                return true;
+            }
+        }
+
+        //Get all .cs files from solution or csproj file
+        /*static public string[] ProjectParser(string projectpath, string filetype)
+        {
+            //TODO TEMP CODE PLEASE IGNORE
+            string[] filelist = new string[1];
+            filelist[0] = "-1";
+            //TODO END TEMP CODE
+            return filelist;
+        } */
+
+        //Get the .CS files contained in the file or directory
+        static public string[] GetCsFiles(string inputFile, string outputFile, bool recurseOption)
+        {
+            string[] filelist;
+            string[] nolist = new string[1];
+            nolist[0] = "-1";
+
+            //Check path
+            if (File.Exists(inputFile))
+            {
+                //TODO TEST CODE
+                // This path is a file
+                Console.WriteLine("FILE");
```

Comment:
The DocNet core will log the current project/file being processed. This can be removed.